Namco system 246/256 support

israpps opened the pull request:
Mostly because ps2ident is complicated to maintain and because we currently don't have any means of dumping the arcade flash memory

israpps committed:
shorter bin2c recipes

israpps committed:
initial incomplete implementation

F0bes commented on biosdrain.c:
```diff
@@ -170,6 +190,7 @@ int main(void)
 	graphic_biosdrain_fadein();
 
 	menu_status("biosdrain - revision %s\n", GIT_VERSION);
+	menu_status("version with namco system 246/256 (COH-H models) support\n");
```
Might want to hide this behind the `SUPPORT_SYSTEM_2x6` define as well

israpps replied:
Forgot about this too.
Will do it when I finish the pr

F0bes commented on irx/arcade_ioprp.img:
Where did this file come from? Was this generated by you?

israpps replied:
Yes, it has homebrew IOMAN and FILEIO inside

F0bes commented on irx/ioprp_img.c:
You don't need to include the bin2c generated files, these are expected to be generated by the makefile rules.

israpps replied:
Yeah, forgot to update gitignore, sorry.

F0bes commented on biosdrain.c:
```diff
@@ -187,3 +208,7 @@ int main(void)
+
+#ifdef SUPPORT_SYSTEM_2x6
+void _libcglue_rtc_update() {} // system 246/256 doesnt have CDVDFSV module loaded on boot
+#endif
```
Add an empty line at the end of this file.

israpps committed:
remove bin2 intermediate file

israpps committed:
do requested changes

israpps committed:
Update Makefile

israpps committed:
make sure arcade syscon doesnt poweroff system

F0bes commented:
Can you verify if the generated artifact works for you on a retail console?
I'm pretty positive it has nothing to do with your changes, but is instead an SDK regression. But it doesn't seem to detect my USB.

israpps commented:
> Can you verify if the generated artifact works for you on a retail console? I'm pretty positive it has nothing to do with your changes, but is instead an SDK regression. But it doesn't seem to detect my USB.

usb should not be affected

On emulator it boots file (with COH-H31100 boot rom)

I could check tomorrow on my SCPH-50009

israpps commented:
I see you are using the vfat driver instead of fatfs. I havent used that one in any of my programs, I always use fatfs.

F0bes commented:
> I see you are using the vfat driver instead of fatfs. I havent used that one in any of my programs, I always use fatfs.

Hmm, good catch. I'll try using fatfs instead.

israpps commented:
> > I see you are using the vfat driver instead of fatfs. I havent used that one in any of my programs, I always use fatfs.
> 
> Hmm, good catch. I'll try using fatfs instead.

I would also recommend the USB wait approach made by fjtrujy
```c
    struct stat buffer;
    int ret = -1;
    int retries = 50;

    while (ret != 0 && retries > 0) {
        ret = stat("mass:/", &buffer);
        /* Wait until the device is ready */
        nopdelay();

        retries--;
    }
```

~edit: also, maybe loading usbd.irx before bdm.irx so we save time while the BDM interface setups~ youre doing this already

israpps committed:
more IOP reboot into an appropiate place

this makes sure CDVD RPC is setup by the time program tries to retrieve console model number

israpps committed:
avoid confusing COH-H and DTL-T with SCPH-10000

israpps committed:
hook arcade variants to CI

F0bes commented:
I'm currently waiting for @fjtrujy to test his ps2_drivers usb sample. I'm currently unable to get it, and using fatfs directly, to work for me.
Once the issue is resolved I'll migrate to fatfs and when you're ready, merge this PR :)

israpps commented:
> I'm currently waiting for @fjtrujy to test his ps2_drivers usb sample. I'm currently unable to get it, and using fatfs directly, to work for me. Once the issue is resolved I'll migrate to fatfs and when you're ready, merge this PR :)

No hurry.
